portablemc/forge.py

```diff
@@ -416,16 +441,38 @@ class ForgePostProcessedEvent:
 
 
 def request_promo_versions() -> Dict[str, str]:
+    """Request recommended and latest versions for each supported game release, this only
+    works.
+    """
     return http_request("GET", "https://files.minecraftforge.net/net/minecraftforge/forge/promotions_slim.json", 
         accept="application/json").json()["promos"]
 
 
-def request_maven_versions() -> List[str]:
+def _request_neoforge_version(game_version: str) -> Optional[str]:
+    """Request a neoforge version from a game version. Because of the NeoForge API 
+    returning any matching version number, this function will also check that the returned
+    version is of the right game version.
+    """
+    try:
+        # NOTE: For now we don't sanitize the parameter.
+        url = f"https://maven.neoforged.net/api/maven/latest/version/releases/net%2Fneoforged%2Fforge?filter={game_version}"
+        ret = http_request("GET", url, accept="application/json").json()
+        loader_version = ret.get("version", "")
+        if not loader_version.startswith(f"{game_version}-"):
+            return None
+        return loader_version
+    except HttpError as err:
+        if err.res.status != 404:
+            raise
+        return None
+
+
+def request_maven_versions(*, _repo: str = _FORGE_REPO) -> List[str]:
     """Internal function that parses maven metadata of forge in order to get all 
     supported forge versions.
     """
 
-    text = http_request("GET", "https://maven.minecraftforge.net/net/minecraftforge/forge/maven-metadata.xml", 
+    text = http_request("GET", f"{_repo}/maven-metadata.xml", 
```

Comment:
Verified unstable API, existing code won't break because it defaults to the same repo